### ISSUE TITLE
Empty data causes other component blocking errors

### ISSUE THREAD

**Issue by Frozenfire92** [Glavin001/ember-c3#9]
If you have an empty data object passed you will get the following error

```
Uncaught Error: url or json or rows or columns is required. 
```

This error will block any other components from rendering properly

**Comment by bkCDL** [comment]
I experienced this too, but found a workaround by couching the chart inside a handlebars IF clause:

    {{#if contentChart}}
            {{c3-chart
                data=contentChart
                axis=chartAxis
                bar=chartBarFormat
                pie=chartPieFormat
                donut=chartDonutFormat
                legend=chartLegendSettings
                tooltip=chartTooltipFormat
                color=chartColors
            }}
    {{/if}}

Bryan

**Comment by JKGisMe** [comment]
Does that mean whatever you put as `data=` has to have a key titled 'columns', 'rows', 'json' or 'url'? I tried feeding it an array of numerical values and I'm getting this error.

**Comment by Glavin001** [comment]
We should add checks on the `data` to determine how it is handled. An empty value for `data` should fail silently, instead of crashing your Ember app or requiring the `if` helper as @bkCDL suggested.
I'll look into this.

**Comment by psteininger** [comment]
@Glavin001 I can confirm that this is an issue. It stops google map from rendering.

**Comment by Glavin001** [comment]
@psteininger the temporary workaround I recommend would be https://github.com/Glavin001/ember-c3/issues/9#issuecomment-71663189
If anyone submits a pull request fixing this issue I will gladly review and merge. Thank you!

**Comment by psteininger** [comment]
@Glavin001 the workaround doesn't work in my case. My data sits in `controller.chart.data` so things don't get messy (I got a config for maps and other stuff there). The issue is happening much earlier. A check for invalid data should happen inside `chart` computed property. I will try and fix it and do a PR if I manage.

**Comment by psteininger** [comment]
this is more of a note-to-self: in https://github.com/Glavin001/ember-c3/blob/master/addon/components/c3-chart.js#L118

I think the cleanest thing to do is wrap things in a try-catch block and re-org things a bit. The reason for it is that C3 already does comprehensive validation, and throws an exception when things aren't kosher. There is no point in doing this twice. we could miss something and still have the same issue under slightly different conditions.

**Comment by psteininger** [comment]
@Glavin001 looks like I will have to tackle this next week. I found another issue - using a nested property just isn't supported now. I would like to add that in as well. I think we should be able to pass a config object, which would be merged with defaults in the components and set up observer on the data attribute of config object.

I won't have time to finish this weekend, but I should be able to swing it on Monday

**Comment by maxwondercorn** [comment]
fixed with pr #86.  Will be released with 1.0.0

**Comment by maxwondercorn** [comment]
See issue #77
